src/main/java/com/hivemq/extensions/packets/connect/ModifiableConnectPacketImpl.java

```diff
@@ -264,10 +288,24 @@ public synchronized void setPassword(final @Nullable ByteBuffer password) {
         if (password == null && this.password == null) {
             return;
         }
+        this.passwordBytes = Bytes.getBytesFromReadOnlyBuffer(Optional.ofNullable(password));
         this.password = password;
         modified = true;
     }
 
+    @Override
+    public void setPassword(@Nullable final byte[] bytes) {
+        if (bytes != null && Arrays.equals(bytes, this.passwordBytes)) {
```

Comment:
Arrays.equals can handle null.
`    /**
     * Returns {@code true} if the two specified arrays of bytes are
     * <i>equal</i> to one another.  Two arrays are considered equal if both
     * arrays contain the same number of elements, and all corresponding pairs
     * of elements in the two arrays are equal.  In other words, two arrays
     * are equal if they contain the same elements in the same order.  Also,
     * two array references are considered equal if both are {@code null}.
     *
     * @param a one array to be tested for equality
     * @param a2 the other array to be tested for equality
     * @return {@code true} if the two arrays are equal
     */`